### PR TITLE
CONFSERVER-94603: Improper Authorization Fix

### PR DESCRIPTION
**Pull request by sumedhabajpai**
PR to fix Improper Authorization

vuln details - https://security.snyk.io/vuln/SNYK-JAVA-COMHAZELCAST-6282853

Backport of following commit to forked library: https://github.com/hazelcast/hazelcast/commit/1eec447afaff529381e8cf8af07b368b2665883c

QA demo for hazelcast upgrade in confluence 8.5 - https://www.loom.com/share/297dab6341384403a8f036f9c2d018b6
QA demo for hazelcast upgrade in confluence 7.19 - https://www.loom.com/share/1c92ffd9ef654365ada2d05ab8b0862d
QA demo for hazelcast upgrade in synchrony 5.1 - https://www.loom.com/share/131b4f4bff9f48cea44849e4041764cc

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/cp/internal/datastructures/semaphore/client/GetSemaphoreTypeMessageTask.java [hunk 1]
```diff
@@ -59,7 +60,7 @@ public String getServiceName() {
 
     @Override
     public Permission getRequiredPermission() {
-        return null;
+        return new SemaphorePermission(parameters.proxyName);
```
Comment (krastogi-atlassian) [inline hazelcast/src/main/java/com/hazelcast/cp/internal/datastructures/semaphore/client/GetSemaphoreTypeMessageTask.java:63]: Shouldn't this be parameters.name?

**Comment by krastogi-atlassian** [comment]
@sumedhabajpai : Did you loked into possibility of adding test?

**Review by krastogi-atlassian — APPROVED** [review]

File: hazelcast/src/main/java/com/hazelcast/cp/internal/datastructures/semaphore/client/GetSemaphoreTypeMessageTask.java (hunk 1 above)
Comment (sumedhabajpai) [reply hazelcast/src/main/java/com/hazelcast/cp/internal/datastructures/semaphore/client/GetSemaphoreTypeMessageTask.java:63]: `parameters.name` is not an available attribute in this context, and `parameters.proxyName` aligns with usage in other methods like `getDistributedObjectName()`.  so I believe using `parameters.proxyName` is the correct approach in this case.

**Comment by sumedhabajpai** [comment]
> @sumedhabajpai : Did you loked into possibility of adding test? [hazelcast@1eec447#diff-70cc42c04eda3ffcf8353133dd31a4b4abd7c23717f5bfa5ddbe15b432376226](https://github.com/hazelcast/hazelcast/commit/1eec447afaff529381e8cf8af07b368b2665883c#diff-70cc42c04eda3ffcf8353133dd31a4b4abd7c23717f5bfa5ddbe15b432376226)

done

**Pull request merged**